test/view/generate_view.cpp

```diff
@@ -0,0 +1,132 @@
+#include <stl2/view/generate_view.hpp>
+
+#include "../simple_test.hpp"
+#include <cstring>
+#include <stl2/ranges.hpp>
+#include <stl2/view/take_exactly.hpp>
+
+namespace ranges = __stl2;
+
+namespace view {
+	using namespace ranges::view;
+	using namespace ext;
+} // namespace view
+
+// a simple type to test move semantics
+struct MoveOnlyString {
+	char const* sz_;
+
+	MoveOnlyString(char const* sz = "")
+	  : sz_(sz)
+	{}
+	MoveOnlyString(MoveOnlyString&& that)
+	  : sz_(that.sz_)
+	{
+		that.sz_ = "";
+	}
+	MoveOnlyString(MoveOnlyString const&) = delete;
+	MoveOnlyString& operator=(MoveOnlyString&& that)
+	{
+		sz_ = that.sz_;
+		that.sz_ = "";
+		return *this;
+	}
+	MoveOnlyString& operator=(MoveOnlyString const&) = delete;
+	bool operator==(MoveOnlyString const& that) const
+	{  return 0 == std::strcmp(sz_, that.sz_); }
+
+	bool operator<(const MoveOnlyString& that) const
+	{ return std::strcmp(sz_, that.sz_) < 0; }
+
+	bool operator!=(MoveOnlyString const& that) const
+	{ return !(*this == that); }
+
+	friend std::ostream& operator<<(std::ostream& sout, MoveOnlyString const& str)
+	{ return sout << '"' << str.sz_ << '"'; }
+};
+
+struct MoveOnlyFunction {
+	MoveOnlyString str_;
+	int i_;
+
+	char operator()()
+	{ return str_.sz_[i_++]; }
+};
+
+int main()
+{
+	// Test for constant generator functions
+	{
+		int i = 0, j = 1;
+		auto fib = view::generate([&]()->int{int tmp = i; i += j; std::swap(i, j); return tmp;});
```

Comment:
```suggestion
		auto fib = view::generate([&]{ int tmp = i; i += j; std::swap(i, j); return tmp; });
```

Reply:
(Meh, I don't actually care for test code, just pointing out the `() -> int` is extraneous.)